src/components/Button.js

```diff
@@ -1,9 +1,46 @@
 import React from 'react';
 
-const ButtonSearch = (props) => (
- <button className="btn btn-primary fn-submit-name">
-     {props.name} {props.id}
-    </button>
-)
+class Button extends React.Component {
+  constructor() {
+    super()
+    this.state = {
+      value: ""
+    }
+  }
+
+  handleUpdate = (e) => {
+    this.setState({ value: e.target.value })
```

Comment:
Because you're listening for updates on the `<input>` and updating the state, this pattern that you're using is called a "controlled component". There's not really a problem with using this pattern, but it is slightly unusual for this particular case. It is often used if we need to transform or keep track of exactly what the user has typed.

React has an alternative pattern (called "uncontrolled component"), were we don't add a `onChange` handler, and then just get the input when we click the button.

I've just realised that we haven't looked a feature of React that will make this easier in class yet (called *refs*). Hopefully that will make more sense on how we can do that after the class.